packages/react-import-remote/src/ImportRemote.tsx

```diff
@@ -21,91 +14,48 @@ export interface Props<Imported = any> {
   defer?: DeferTiming;
 }
 
-interface State {
-  loaded: boolean;
-  loading: boolean;
-}
-
-export default class ImportRemote extends React.PureComponent<Props, State> {
-  state: State = {loaded: false, loading: false};
-  private idleCallbackHandle?: RequestIdleCallbackHandle;
-
-  componentWillUnmount() {
-    if (this.idleCallbackHandle != null && 'cancelIdleCallback' in window) {
-      (window as WindowWithRequestIdleCallback).cancelIdleCallback(
-        this.idleCallbackHandle,
-      );
-    }
-  }
-
-  async componentDidMount() {
-    const {defer = DeferTiming.Mount} = this.props;
-
-    if (defer === DeferTiming.Idle) {
-      if ('requestIdleCallback' in window) {
-        this.idleCallbackHandle = (window as WindowWithRequestIdleCallback).requestIdleCallback(
-          this.loadRemote,
-        );
-      } else {
-        this.loadRemote();
+export default function ImportRemote({
+  source,
+  nonce,
+  preconnect,
+  onError,
+  getImport,
+  onImported,
+  defer,
+}: Props) {
+  const {result, intersectionRef} = useImportRemote(source, {
```

Comment:
We should make `ImportRemote` generic, and pass that generic through to the props and here.